web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/SingleUseKeyService.java

```diff
@@ -0,0 +1,105 @@
+package edu.unc.lib.boxc.web.services.processing;
+
+import edu.unc.lib.boxc.model.api.exceptions.RepositoryException;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.UUID;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static edu.unc.lib.boxc.web.services.utils.CsvUtil.createCsvPrinter;
+import static edu.unc.lib.boxc.web.services.utils.CsvUtil.createNewCsvPrinter;
+import static edu.unc.lib.boxc.web.services.utils.CsvUtil.parseCsv;
+
+/**
+ * Generate and invalidate access keys for single use links
+ * @author snluong
+ */
+public class SingleUseKeyService {
+    public static final String ID = "UUID";
+    public static final String ACCESS_KEY = "Access Key";
+    public static final String TIMESTAMP = "Expiration Timestamp";
+    public static final String[] CSV_HEADERS = new String[] {ID, ACCESS_KEY, TIMESTAMP};
+    public static final long DAY_MILLISECONDS = 86400000;
+    private Path csvPath;
+    private ReentrantLock lock = new ReentrantLock();
+
+    /**
+     * Generates an access key for a particular ID, adds it to the CSV, and returns the key
+     * @param id UUID of the record
+     * @return generated access key
+     */
+    public String generate(String id) {
+        var key = getKey();
+        var expirationInMilliseconds = System.currentTimeMillis() + DAY_MILLISECONDS;
+        lock.lock();
+        try (var csvPrinter = createCsvPrinter(CSV_HEADERS, csvPath)) {
+            csvPrinter.printRecord(id, key, expirationInMilliseconds);
+        } catch (Exception e) {
+            throw new RepositoryException("Failed to write new key to Single Use Key CSV", e);
+        } finally {
+            lock.unlock();
+        }
+        return key;
+    }
+
+    /**
+     * Determines if a key is valid by seeing if it is in the CSV and if the expiration timestamp has not passed
+     * @param key access key for single use link
+     * @return true if key is in the CSV, otherwise false
+     * @throws IOException
+     */
+    public boolean keyIsValid(String key) throws IOException {
+        var csvRecords = parseCsv(CSV_HEADERS, csvPath);
```

Comment:
since the other major public methods in this class all throw RepositoryException when an IOException is thrown, it would be consistent to do the same on this line and remove the throws from this method